src/OpenTelemetry.Collector.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -0,0 +1,124 @@
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher dataSourceFetcher = new PropertyFetcher("DataSource");
+        private readonly PropertyFetcher databaseFetcher = new PropertyFetcher("Database");
+        private readonly PropertyFetcher commandTextFetcher = new PropertyFetcher("CommandText");
+        private readonly PropertyFetcher exceptionFetcher = new PropertyFetcher("Exception");
+
+        public SqlClientDiagnosticListener(string sourceName, ITracer tracer)
+            : base(sourceName, tracer)
+        {
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+        }
+
+        public override void OnCustom(string name, Activity activity, object payload)
+        {
+            switch (name)
+            {
+                case "System.Data.SqlClient.WriteCommandBefore":
+                case "Microsoft.Data.SqlClient.WriteCommandBefore":
+                    {
+                        var command = this.commandFetcher.Fetch(payload);
+
+                        const string EventNameSuffix = ".OnStartActivity";
+                        if (command == null)
+                        {
+                            CollectorEventSource.Log.NullPayload(nameof(SqlClientDiagnosticListener) + EventNameSuffix);
+                            return;
+                        }
+
+                        var connection = this.connectionFetcher.Fetch(command);
+                        var dataSource = this.dataSourceFetcher.Fetch(connection);
+                        var database = this.databaseFetcher.Fetch(connection);
+                        var commandText = this.commandTextFetcher.Fetch(command);
+
+                        this.Tracer.StartActiveSpan(database.ToString(), SpanKind.Client, out var span);
+
+                        span.PutComponentAttribute("SQL Server");
+
+                        span.PutDatabaseTypeAttribute("sql");
+                        span.PutDatabaseInstanceAttribute(database.ToString());
+                        span.PutDatabaseStatementAttribute(commandText.ToString());
+                        span.SetAttribute("peer.address", dataSource.ToString());
+                    }
+
+                    break;
+                case "System.Data.SqlClient.WriteCommandAfter":
+                case "Microsoft.Data.SqlClient.WriteCommandAfter":
+                    {
+                        const string EventNameSuffix = ".OnStopActivity";
+                        var span = this.Tracer.CurrentSpan;
+
+                        if (span == null || span == BlankSpan.Instance)
+                        {
+                            CollectorEventSource.Log.NullOrBlankSpan(nameof(SqlClientDiagnosticListener) + EventNameSuffix);
```

Comment:
```suggestion
                            CollectorEventSource.Log.NullOrBlankSpan(nameof(SqlClientDiagnosticListener) + name);
```

no stop activity event for SQL